src/main.cpp

```diff
@@ -3631,10 +3612,15 @@ bool CBlock::SetBestChain(CTxDB& txdb, CBlockIndex* pindexNew)
                 printf("SetBestChain() : TxnBegin 2 failed\n");
                 break;
             }
+
             // errors now are not fatal, we still did a reorganisation to a new chain in a valid way
             if (!block.SetBestChainInner(txdb, pindex, true))
                 break;
         }
+
+        // Retally after reorganize to sync up amounts owed.
+        BusyWaitForTally_retired();
+        TallyNetworkAverages_v9();
```

Comment:
Please also have a look at [main.cpp#L3431](https://github.com/denravonska/Gridcoin-Research/blob/6875db3d52d39ae2f26f0cab202999ff2dce3aae/src/main.cpp#L3431) `GetLifetimeCPID()` call, because it does not work properly. It needs to be here, where memory index is consistent, an not where it is currently.

Reply:
Will do!

Reply:
Started adding this but we either need to remove the disconnected payments from `StructCPID::PaymentAmountsResearch` (and its siblings) or we remove the variables entirely. They are only used in the CSV magnitude report and serve no other purpose.